src/metabase/search/filter.clj

```diff
@@ -129,35 +129,63 @@
-                               (throw (ex-info (tru "Failed to parse created at param: {0}" created-at) {:status-code 400}))))
-        start               (some-> start u.date/parse)
-        end                 (some-> end u.date/parse)
-        created-at-col      (search.config/column-with-model-alias model :created_at)
-        created-at-col      (if (some #(instance? LocalDate %) [start end])
-                             [:cast created-at-col :date]
-                             created-at-col)]
+(defn- date-range-filter-clause
+  [dt-col dt-val]
+  (let [date-range (try
+                    (params.dates/date-string->range dt-val {:inclusive-end? false})
+                    (catch Exception _e
+                      (throw (ex-info (tru "Failed to parse datetime value: {0}" dt-val) {:status-code 400}))))
+        start      (some-> (:start date-range) u.date/parse)
+        end        (some-> (:end date-range) u.date/parse)
+        dt-col     (if (some #(instance? LocalDate %) [start end])
+                     [:cast dt-col :date]
+                     dt-col)]
     (cond
      (= start end)
-     [:= created-at-col start]
+     [:= dt-col start]
 
      (nil? start)
-     [:< created-at-col end]
+     [:< dt-col end]
 
      (nil? end)
-     [:> created-at-col start]
+     [:> dt-col start]
 
      :else
-     [:and [:>= created-at-col start] [:< created-at-col end]])))
+     [:and [:>= dt-col start] [:< dt-col end]])))
 
 (doseq [model ["collection" "database" "table" "dashboard" "card" "dataset" "action"]]
   (defmethod build-optional-filter-query [:created-at model]
     [_filter model query created-at]
-    (sql.helpers/where query (default-created-at-filter-clause model created-at))))
+    (sql.helpers/where query (date-range-filter-clause
+                              (search.config/column-with-model-alias model :created_at)
+                              created-at))))
+
+
+;; Last edited by filter
+
+(defn- search-model->revision-model
+  [model]
+  (case model
+    "dataset" (recur "card")
+    (str/capitalize model)))
+
+(doseq [model ["dashboard" "card" "dataset" "metric"]]
+  (defmethod build-optional-filter-query [:last-edited-by model]
+    [_filter model query last-edited-by]
+    (-> query
+        (sql.helpers/join :revision
+                          [:= :revision.model_id
+                           (search.config/column-with-model-alias model :id)])
+        (sql.helpers/where [:= :revision.most_recent true]
+                           [:= :revision.model (search-model->revision-model model)]
+                           [:= :revision.user_id last-edited-by]))))
+
+(doseq [model ["action" "dashboard" "card" "dataset" "metric"]]
+  (defmethod build-optional-filter-query [:last-edited-at model]
+    [_filter model query last-edited-at]
+    (sql.helpers/where query (date-range-filter-clause
+                              (search.config/column-with-model-alias model :updated_at)
```

Comment:
This should be `revision.timestamp`, not the `updated_at` timestamp. We display use the revision timestamp throughout the app with `LastEditInfo`, rather than `updated_at`. I believe it is because there are ways something can be updated, but not qualify as "edits". We should be consistent

See 
https://github.com/metabase/metabase/blob/c40185a8d036248875e6723210d29db136a5c1ab/frontend/src/metabase/collections/components/BaseTableItem.jsx#L144-L148
or 
https://github.com/metabase/metabase/blob/c40185a8d036248875e6723210d29db136a5c1ab/frontend/src/metabase/query_builder/components/dataref/QuestionPane/QuestionPane.tsx#L82-L89

Reply:
ah, that's a good find. will change!

Reply:
changed to use `revision.timestamp` for filtering dashboard,card,metric.

for action we still need to use `updated_at` column because we haven't recorded revision for action yet.

https://github.com/metabase/metabase/pull/33405/commits/91b0bf3d964a9c851e036d8b7bf462461d66cab1